### PR TITLE
Clarify IMAP character set

### PR DESCRIPTION
**Pull request by tkdrob**
## Proposed change

I use Yahoo Mail and noticed that it needed to use a different character set to get it working. This might be done better in the integration but always good to document where someone might go wrong and how to fix it.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/imap.markdown [hunk 1]
```diff
@@ -40,15 +40,19 @@ By default, this integration will count unread emails. By configuring the search
 * `FROM`, `TO`, `SUBJECT` to find emails in a folder (see [IMAP RFC for all standard options](https://tools.ietf.org/html/rfc3501#section-6.4.4))
 * [Gmail's IMAP extensions](https://developers.google.com/gmail/imap/imap-extensions) allow raw Gmail searches, like `X-GM-RAW "in: inbox older_than:7d"` to show emails older than one week in your inbox. Note that raw Gmail searches will ignore your folder configuration and search all emails in your account!
 
+If you run into an error saying the selected search is invalid, that could be the case or you need to use a different character set.
```
Comment (dannytsang) [inline source/_integrations/imap.markdown]: The error and possible solution seems too vague in my opinion. I would leave this bit out and keep your note below on setting the character set for Yahoo.

File: source/_integrations/imap.markdown [hunk 2]
```diff
@@ -40,15 +40,19 @@ By default, this integration will count unread emails. By configuring the search
...
 
-Below is an example for setting up the integration to connect to your Microsoft 365 account that requires `US_ASCII` as charset:
+Below is an example for setting up the integration to connect to your Microsoft 365 account that requires `US-ASCII` as charset:
   - Server: `outlook.office365.com`
   - Port: `993`
   - Username: Your full email address
   - Password: Your password
   - Charset: `US-ASCII`
 
+Note that Yahoo also requires the character set `US-ASCII`.
```
Comment (dannytsang) [inline source/_integrations/imap.markdown]: Use the `<div class="note"></div>` so it styles this paragraph.

**Review by dannytsang — DISMISSED** [review]

File: source/_integrations/imap.markdown (hunk 1 above)
Comment (c0ffeeca7) [inline source/_integrations/imap.markdown]:
```suggestion

```

File: source/_integrations/imap.markdown (hunk 2 above)
Comment (c0ffeeca7) [inline source/_integrations/imap.markdown]:
```suggestion
<div class="note">
Note that Yahoo also requires the character set `US-ASCII`.
</div>
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]

File: source/_integrations/imap.markdown [hunk 3]
```diff
@@ -41,15 +41,21 @@ By default, this integration will count unread emails. By configuring the search
...
 
-Below is an example for setting up the integration to connect to your Microsoft 365 account that requires `US_ASCII` as charset:
+Below is an example for setting up the integration to connect to your Microsoft 365 account that requires `US-ASCII` as charset:
   - Server: `outlook.office365.com`
   - Port: `993`
   - Username: Your full email address
   - Password: Your password
   - Charset: `US-ASCII`
 
+
```
Comment (c0ffeeca7) [inline source/_integrations/imap.markdown]:
```suggestion
```

File: source/_integrations/imap.markdown [hunk 4]
```diff
@@ -41,15 +41,20 @@ By default, this integration will count unread emails. By configuring the search
...
-Below is an example for setting up the integration to connect to your Microsoft 365 account that requires `US_ASCII` as charset:
+Below is an example for setting up the integration to connect to your Microsoft 365 account that requires `US-ASCII` as charset:
   - Server: `outlook.office365.com`
   - Port: `993`
   - Username: Your full email address
   - Password: Your password
   - Charset: `US-ASCII`
 
+<div class="note">
+Note that Yahoo also requires the character set `US-ASCII`.
```
Comment (c0ffeeca7) [inline source/_integrations/imap.markdown]:
```suggestion
Yahoo also requires the character set `US-ASCII`.
```

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @tkdrob :+1:

File: source/_integrations/imap.markdown (hunk 1 above)
Comment (c0ffeeca7) [reply source/_integrations/imap.markdown]: removed the note

**Comment by c0ffeeca7** [comment]
Implemented the review comments to move this along.

File: source/_integrations/imap.markdown [hunk 5]
```diff
@@ -41,15 +41,20 @@ By default, this integration will count unread emails. By configuring the search
...
+Below is an example for setting up the integration to connect to your Microsoft 365 account that requires `US-ASCII` as charset:
   - Server: `outlook.office365.com`
   - Port: `993`
   - Username: Your full email address
   - Password: Your password
   - Charset: `US-ASCII`
 
+<div class="note">
+Yahoo also requires the character set `US-ASCII`.
+</div>
```
Comment (frenck) [inline source/_integrations/imap.markdown]:
```suggestion
<div class="note">

Yahoo also requires the character set `US-ASCII`.

</div>
```

**Review by frenck — APPROVED** [review]

**Review by jbouwh — APPROVED** [review]
LGTM.
Thnx @tkdrob 👍

**Pull request merged**